libopenthread/platform/plat.h

```diff
@@ -0,0 +1,12 @@
+#ifndef PLAT_H
+#define PLAT_H
+#include<stdbool.h>
+
+typedef struct {
+    char *buffer;
+    int write_index;
+    int read_index;
+    bool new;
+} ring_buffer;
```

Comment:
What is this for?

Reply:
This is being used to solve the race conditions and problems we faced with receive buffers and the receive upcall. 

There is now a ring buffer in the capsule and also a ring buffer in the openthread app. The app's ring buffer performs a sync copy from the allowed kernel ring buffer and completes the entire operation before passing the packet to openthread (some of openthread's operations may yield which was causing numerous issues with handling packets being passed to the process). 

This userprocess ring buffer could be removed in the future with a kernel allow ring buffer syscall.